README.md

```diff
@@ -111,170 +106,53 @@ See [the rustc-dev-guide for more info][sysllvm].
 
-4. Navigate to Rust's source code (or clone it), then build it:
+Rust programs which use cargo can now be built and run using Alloy instead of
+the official Rust compiler:
 
    ```sh
-   python x.py setup user && python x.py build && python x.py install
+   cargo +alloy build
    ```
 
-#### MSVC
-
-MSVC builds of Rust additionally require an installation of Visual Studio 2017
-(or later) so `rustc` can use its linker. The simplest way is to get
-[Visual Studio], check the "C++ build tools" and "Windows 10 SDK" workload.
-
-[Visual Studio]: https://visualstudio.microsoft.com/downloads/
-
-(If you're installing CMake yourself, be careful that "C++ CMake tools for
-Windows" doesn't get included under "Individual components".)
-
-With these dependencies installed, you can build the compiler in a `cmd.exe`
-shell with:
-
-```sh
-python x.py setup user
-python x.py build
-```
-
-Right now, building Rust only works with some known versions of Visual Studio.
-If you have a more recent version installed and the build system doesn't
-understand, you may need to force rustbuild to use an older version.
-This can be done by manually calling the appropriate vcvars file before running
-the bootstrap.
-
-```batch
-CALL "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"
-python x.py build
-```
-
-#### Specifying an ABI
+## How it works
 
-Each specific ABI can also be used from either environment (for example, using
-the GNU ABI in PowerShell) by using an explicit build triple. The available
-Windows build triples are:
-- GNU ABI (using GCC)
-    - `i686-pc-windows-gnu`
-    - `x86_64-pc-windows-gnu`
-- The MSVC ABI
-    - `i686-pc-windows-msvc`
-    - `x86_64-pc-windows-msvc`
+[Boehm Demers Weiser GC (BDWGC)]: https://github.com/ivmai/bdwgc
 
-The build triple can be specified by either specifying `--build=<triple>` when
-invoking `x.py` commands, or by creating a `config.toml` file (as described in
-[Building on a Unix-like system](#building-on-a-unix-like-system)), and passing
-`--set build.build=<triple>` to `./configure`.
+Alloy uses _conservative_ garbage collection. This means that it does not have
+any specific knowledge about where references to objects are located. Instead,
+Alloy will assume that an object is still alive if it can be reached by a value on
+the stack (or in a register) which, if treated like a pointer, points to an
+object in the heap. The fields of those objects are then traced using the same
+approach, until all live objects in the program have been discovered. 
 
-## Building Documentation
+This tends to work well in practice, however, it comes with an important caveat:
+you must not hide references from the type system. For example, data structures
+such as XOR lists are unsound because Alloy will never be able to reach their
+objects.
 
-If you'd like to build the documentation, it's almost the same:
+Behind the scenes, Alloy uses the [Boehm Demers Weiser GC (BDWGC)] for its
```

Comment:
Missing URL in brackets for BDWGC?

Reply:
No this is just weird Markdown syntax. The link is on line +124

Reply:
Oh!